onnx/defs/nn/defs.cc

```diff
@@ -3330,6 +3330,8 @@ Attention bias to be added is calculated based on `attn_mask` input and `is_caus
 1) If `is_causal` is set to `1`, the attention masking is a lower triangular matrix when the mask is a square matrix. The attention masking has the form of the upper left causal bias due to the alignment.
 2) `attn_mask`: A boolean mask where a value of `True` indicates that the element should take part in attention or a float mask of the same type as query, key, value that is added to the attention score.
 
+Optionally, `kv_nonpad_seqlen` can be provided to indicate the number of valid tokens in each sample.
```

Comment:
Please add a description of how this handled. Clarify that it is of shape `(batchsize,)` and that it is used only to construct a corresponding causal-mask. Clarify that it can't be used with user-provided attention-mask input. I guess the causal-mask is redundant if this input is provided?